pulsar-client-cpp/lib/ConnectionPool.cc

```diff
@@ -37,6 +45,27 @@ Future<Result, ClientConnectionWeakPtr> ConnectionPool::getConnectionAsync(
     const std::string& logicalAddress, const std::string& physicalAddress) {
     boost::unique_lock<boost::mutex> lock(mutex_);
 
+    if (clientConfiguration_.isValidateHostName()) {
+        // Create a context that uses the default paths for
+        // finding CA certificates.
+        ssl::context ctx(ssl::context::sslv23);
+        ctx.set_default_verify_paths();
+
+        // Open a socket and connect it to the remote host.
+        boost::asio::io_service io_service;
+        ssl_socket sock(io_service, ctx);
+        tcp::resolver resolver(io_service);
+        tcp::resolver::query query(physicalAddress, "https");
+        boost::asio::connect(sock.lowest_layer(), resolver.resolve(query));
+        sock.lowest_layer().set_option(tcp::no_delay(true));
+
+        // Perform SSL handshake and verify the remote host's
+        // certificate.
+        sock.set_verify_mode(ssl::verify_peer);
+        sock.set_verify_callback(ssl::rfc2818_verification(physicalAddress));
+        sock.handshake(ssl_socket::client);
```

Comment:
is it possible to add unit-test? we can reuse certificates used in [java-test](https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/test/java/org/apache/pulsar/client/api/AuthenticationTlsHostnameVerificationTest.java)

Reply:
I will add unit test first I need to fix the build